pkg/secrethub/credentials/providers.go

```diff
@@ -60,7 +52,7 @@ func UseAWS(awsCfg ...*awssdk.Config) Provider {
 //		credentials.UseKey(credentials.FromBytes("<a credential>"))
 //		credentials.UseKey(credentials.FromFile("~/.secrethub/credential"), credentials.FromString("passphrase"))
 func UseKey(credentialReader Reader, passReader Reader) Provider {
-	return func() (http.AuthProvider, Decrypter, error) {
+	return func(_ *http.Client) (auth.Authenticator, Decrypter, error) {
```

Comment:
Did we just move the problem one layer up but it's still there?

Reply:
I challenge you to solve it differently.